test/protobuf/decoder_test.exs

```diff
@@ -126,7 +126,7 @@ defmodule Protobuf.Decoder.Test do
       use Protobuf, from: Path.expand("../proto/mumble.proto", __DIR__)
     end
 
-    msg = MumbleProto.Authenticate.new(username: "bitwalker")
+    msg = MumbleProto.MumbleProto.Authenticate.new(username: "bitwalker")
```

Comment:
I haven't looked through the code in depth yet, but how are you avoiding defining the module inside of `NoTopLevel` here? It seems to me that no matter how you define a module inside of `__using__`, you can't escape it's context. Though maybe manipulating the quoted terms would allow you to replace the context. Anyway, yeah that's definitely different behaviour from what exists today. It also presents a problem, in that if you happen to have a protobuf message with the same name as a module in your application, the Elixir standard library, or a dependency, there will be a conflict (and I'm fairly sure the module defined by exprotobuf would redefine whatever originally had the name).

The purpose of `inject` was two-fold, 1.) it allowed you to name a protobuf message whatever you wanted within the context of your application, and 2.) made the namespacing of messages injected explicit. I kind of feel like it's important that the modules injected have a clear root namespace, or it could be really confusing if you run into the conflict scenario I outlined above. Mix will warn you on compile that it's being redefined, but that doesn't tell you why, or how.

Thoughts?